here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -64,9 +67,17 @@ public SourceIdHandler(
   }
 
   @Override
-  public @NotNull Result processEvent(@NotNull IEvent event) {
+  protected EventProcessingStrategy processingStrategyFor(IEvent event) {
+    Request<?> request = event.getRequest();
+    if (request instanceof ReadFeatures || request instanceof WriteXyzFeatures) {
+      return PROCESS;
+    }
+    return SEND_UPSTREAM_WITHOUT_PROCESSING;
+  }
 
-    final Request<?> request = event.getRequest();
+  @Override
+  public @NotNull Result process(@NotNull IEvent event) {
+    Request<?> request = event.getRequest();
```

Comment:
The reason why it was important to declare the "request" as final (in previous code), was to ensure that we manipulate request in this handler and forward the same request object to next handler (and don't create a new object). Creating new request may result in loss of data, if let's say incoming request was not a direct ReadFeatures but may be subclass of ReadFeatures having some additional inputs in it.
So, removing final will still work for now (and is not a functional problem), but having it final makes someone think before creating a new request object altogether in future (well, I know that is obvious explanation of using a final keyword :D but it was the same reason here). So, unless there is specific reason, let's continue with it being a final.

Reply:
Actually, it was not my intention to "unfinalize" this, I must have changed it accidentally - good that you spotted  it :) I don't like mutability & variability as it opens up the whole category of problems - I will make this final again

Reply:
Ok, being unintentional is less of a worry then, as there is no conflict in understanding ;)